pkg/haproxy/haproxy_test.go

```diff
@@ -151,6 +151,56 @@ func TestCreateLoadBalancer(t *testing.T) {
 				IP:       http2Addr,
 				NodeName: pointer.StringPtr("http2"),
 			},
+		}).
+		WithExtraServices([]haproxy.Frontend{
+			{
+				Name: "test-extra-service-1",
+				Port: uint32(RandomTCPPort()),
+				Backend: &haproxy.Backend{
+					Port: uint32(RandomTCPPort()),
+					Addresses: []corev1.EndpointAddress{
+						{
+							IP:       http1Addr,
+							NodeName: pointer.StringPtr("http1-extra-service-1"),
+						},
+						{
+							IP:       http2Addr,
+							NodeName: pointer.StringPtr("http2-extra-service-1"),
+						},
+					},
+					HealthCheckConfig: &haproxy.BackendHealthCheckConfig{
+						BackendHealthCheckOption: haproxy.BackendHealthCheckOption{
+							HTTPCheck: &haproxy.HTTPCheck{
+								Method: "GET",
+								URI:    "/readyz",
+								SSL:    true,
+								Response: &haproxy.HTTPCheckResponse{
+									HTTPCheckResponseOption: haproxy.HTTPCheckResponseOption{
+										Status: "200",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			{
+				Name: "test-extra-service-2",
+				Port: uint32(RandomTCPPort()),
+				Backend: &haproxy.Backend{
+					Port: uint32(RandomTCPPort()),
+					Addresses: []corev1.EndpointAddress{
+						{
+							IP:       http1Addr,
+							NodeName: pointer.StringPtr("http1-extra-service-2"),
+						},
+						{
+							IP:       http2Addr,
+							NodeName: pointer.StringPtr("http2-extra-service-2"),
+						},
+					},
+				},
+			},
```

Comment:
Are we asserting the rendered config against something?

Reply:
It was only verifying if config update is successful. But added assertion now